framework/webtools/groovyScripts/entity/XmlDsDump.groovy

```diff
@@ -17,151 +17,158 @@
  * under the License.
  */
 
-import java.util.*
-import java.io.*
-import org.apache.ofbiz.base.util.*
-import org.apache.ofbiz.entity.model.*
-import org.apache.ofbiz.entity.util.*
-import org.apache.ofbiz.entity.transaction.*
-import org.apache.ofbiz.entity.condition.*
+import org.apache.ofbiz.base.util.Debug
+import org.apache.ofbiz.base.util.UtilFormatOut
+import org.apache.ofbiz.entity.condition.EntityComparisonOperator
+import org.apache.ofbiz.entity.condition.EntityCondition
+import org.apache.ofbiz.entity.condition.EntityJoinOperator
+import org.apache.ofbiz.entity.model.ModelViewEntity
+import org.apache.ofbiz.entity.transaction.TransactionUtil
+import org.apache.ofbiz.entity.util.EntityFindOptions
 
 outpath = parameters.outpath
 filename = parameters.filename
 maxRecStr = parameters.maxrecords
 entitySyncId = parameters.entitySyncId
 passedEntityNames = null
-if (parameters.entityName) passedEntityNames = parameters.entityName instanceof Collection ? parameters.entityName as TreeSet : [parameters.entityName] as TreeSet
+if (parameters.entityName) {
+    passedEntityNames = parameters.entityName instanceof Collection ? parameters.entityName as TreeSet : [parameters.entityName] as TreeSet
+}
 
 // get the max records per file setting and convert to a int
-maxRecordsPerFile = 0
-if (maxRecStr) {
-    try {
-        maxRecordsPerFile = Integer.parseInt(maxRecStr)
-    }
-    catch (Exception e) {
-    }
-}
+int maxRecordsPerFile = maxRecStr ? (maxRecStr as int) : 0
```

Comment:
I agree with your suggestion, thanks

Reply:
I don't think this should have been resolved - as far as I can tell the potential error still exists on line 39, but would not have occurred in the code used prior to this PR.

Reply:
maxRecStr is an user filled data for `Max Records Per File` during xml export. If the user to not fill an integer, this code won't break and maxRecordsPerFile will still be set at default `0` value.
Are you telling that there are some cases that could throw an exception with `isInteger` test to int cast ?

Reply:
No, I'm saying that the use of isInteger code is missing from the PR at this time. 

Once the suggested change, `&& maxRecStr.isInteger()`, has been committed to this PR then this review can be resolved.

Reply:
it was in https://github.com/apache/ofbiz-framework/pull/517/commits/f58a85cb16f85bf5ceab0b9643abc365f03189c2

Reply:
My apologies. 
GitHub was showing me an older version of the file, probably because I clicked on the link associated with this review, rather than navigating to the file from the Files Changed tab.

I can now see the updated version.